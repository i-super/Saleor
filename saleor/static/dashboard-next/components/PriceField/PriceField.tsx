import FormControl from "@material-ui/core/FormControl";
import FormHelperText from "@material-ui/core/FormHelperText";
import Input from "@material-ui/core/Input";
import InputAdornment from "@material-ui/core/InputAdornment";
import InputLabel from "@material-ui/core/InputLabel";
import { withStyles } from "@material-ui/core/styles";
import * as React from "react";

interface PriceRangeFieldProps {
  currencySymbol: string;
  disabled?: boolean;
  error?: boolean;
  hint?: string;
  label?: string;
  name: string;
  value: {
    max: string | number;
    min: string | number;
  };
  onChange(event: any);
}

const decorate = withStyles(theme => ({
  currencySymbol: {
    fontSize: "0.875rem"
  },
  inputContainer: {
    display: "grid",
    gridTemplateColumns: "1fr 2rem 1fr"
  },
  pullDown: {
    marginTop: theme.spacing.unit * 2
  },
  separator: {
    marginTop: theme.spacing.unit * 3,
    textAlign: "center" as "center",
    width: "100%"
  },
  widgetContainer: {
    marginTop: theme.spacing.unit * 2
  }
}));

export const PriceRangeField = decorate<PriceRangeFieldProps>(
  ({
    disabled,
    error,
    label,
    hint,
    currencySymbol,
    name,
    classes,
    onChange,
    value
  }) => (
    <div className={classes.widgetContainer}>
      <div className={classes.inputContainer}>
        <FormControl error={error}>
          {label && <InputLabel htmlFor={`${name}_min`}>{label}</InputLabel>}
          <Input
            value={value ? value.min : ""}
            endAdornment={
              currencySymbol ? (
                <InputAdornment position="end">{currencySymbol}</InputAdornment>
              ) : (
                <span />
              )
            }
            className={label ? "" : classes.pullDown}
            fullWidth
            name={`${name}_min`}
            onChange={onChange}
            type="number"
            disabled={disabled}
          />
          {hint && <FormHelperText>{hint}</FormHelperText>}
        </FormControl>
        <span className={classes.separator}>-</span>
        <FormControl error={error}>
          <Input
            value={value ? value.max : ""}
            endAdornment={
              currencySymbol ? (
                <InputAdornment position="end">{currencySymbol}</InputAdornment>
              ) : (
                <span />
              )
            }
            className={classes.pullDown}
            fullWidth
            name={`${name}_max`}
            onChange={onChange}
            type="number"
            disabled={disabled}
          />
        </FormControl>
      </div>
    </div>
  )
);
PriceRangeField.defaultProps = {
  name: "price"
};

interface PriceFieldProps {
  currencySymbol?: string;
  disabled?: boolean;
  error?: boolean;
  hint?: string;
  label?: string;
  name?: string;
  value?: string | number;
  onChange(event: any);
}

export const PriceField = decorate<PriceFieldProps>(
  ({
    disabled,
    error,
    label,
    hint,
    currencySymbol,
    name,
    classes,
    onChange,
    value
  }) => (
    <FormControl error={error} fullWidth>
      {label && <InputLabel htmlFor={name}>{label}</InputLabel>}
      <Input
        value={value || ""}
        endAdornment={
          currencySymbol ? (
            <InputAdornment position="end" className={classes.currencySymbol}>
              {currencySymbol}
            </InputAdornment>
          ) : (
            <span />
          )
        }
        name={name}
        onChange={onChange}
        type="number"
        disabled={disabled}
      />
      {hint && <FormHelperText>{hint}</FormHelperText>}
    </FormControl>
  )
);
PriceField.defaultProps = {
  name: "price"
};

export default PriceField;
