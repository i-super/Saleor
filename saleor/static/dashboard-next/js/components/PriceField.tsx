import { FormControl, FormHelperText } from "material-ui/Form";
import Input, { InputAdornment, InputLabel } from "material-ui/Input";
import { withStyles } from "material-ui/styles";
import * as React from "react";

interface PriceFieldProps {
  currencySymbol: string;
  hint?: string;
  label: string;
  name: string;
  value: {
    max?: string;
    min?: string;
  };
  onChange(event: any);
}

const decorate = withStyles(theme => ({
  inputContainer: {
    display: "grid",
    gridTemplateColumns: "calc(50% - 1rem) 2rem calc(50% - 1rem)"
  },
  maxInput: {
    marginTop: theme.spacing.unit * 2
  },
  separator: {
    marginTop: theme.spacing.unit * 3,
    textAlign: "center",
    width: "100%"
  },
  widgetContainer: {
    marginTop: theme.spacing.unit * 2
  }
}));

export const PriceField = decorate<PriceFieldProps>(
  ({ label, hint, currencySymbol, name, classes, onChange, value }) => (
    <div className={classes.widgetContainer}>
      <div className={classes.inputContainer}>
        <FormControl>
          <InputLabel htmlFor={`${name}_min`}>{label}</InputLabel>
          <Input
            value={value.min}
            endAdornment={
              <InputAdornment position="end">{currencySymbol}</InputAdornment>
            }
            fullWidth
            name={`${name}_min`}
            onChange={onChange}
            type="number"
          />
        </FormControl>
        <span className={classes.separator}>-</span>
        <FormControl>
          <Input
            value={value.max}
            endAdornment={
              <InputAdornment position="end">{currencySymbol}</InputAdornment>
            }
            className={classes.maxInput}
            fullWidth
            name={`${name}_max`}
            onChange={onChange}
            type="number"
          />
        </FormControl>
      </div>
      {hint && <FormHelperText>{hint}</FormHelperText>}
    </div>
  )
);

export default PriceField;
