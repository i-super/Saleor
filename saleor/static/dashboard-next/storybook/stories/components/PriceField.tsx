import { storiesOf } from "@storybook/react";
import * as React from "react";

import PriceField from "../../../components/PriceField";

const value = {
  max: "30",
  min: "10"
};

storiesOf("Generics / PriceField", module)
  .add("with no value", () => <PriceField onChange={() => {}} />)
  .add("with value", () => <PriceField value={value} onChange={() => {}} />)
  .add("with label", () => (
    <PriceField label="Lorem ipsum" onChange={() => {}} />
  ))
  .add("with hint", () => <PriceField hint="Lorem ipsum" onChange={() => {}} />)
  .add("with label and hint", () => (
    <PriceField label="Lorem" hint="Ipsum" onChange={() => {}} />
  ))
  .add("with value, label and hint", () => (
    <PriceField value={value} label="Lorem" hint="Ipsum" onChange={() => {}} />
  ))
  .add("with value, label and error", () => (
    <PriceField
      value={value}
      label="Lorem"
      hint="Ipsum"
      error={true}
      onChange={() => {}}
    />
  ));
