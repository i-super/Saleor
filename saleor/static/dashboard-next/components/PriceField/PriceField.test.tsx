import * as React from "react";
import * as renderer from "react-test-renderer";

import PriceField from "./";

const value = {
  max: "30",
  min: "10"
};

describe("<PriceField />", () => {
  it("renders with no value", () => {
    const component = renderer.create(<PriceField onChange={() => {}} />);
    expect(component).toMatchSnapshot();
  });
  it("renders with value", () => {
    const component = renderer.create(
      <PriceField value={value} onChange={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with label", () => {
    const component = renderer.create(
      <PriceField label="Lorem ipsum" onChange={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with hint", () => {
    const component = renderer.create(
      <PriceField hint="Lorem ipsum" onChange={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with label and hint", () => {
    const component = renderer.create(
      <PriceField label="Lorem" hint="Ipsum" onChange={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with value, label and hint", () => {
    const component = renderer.create(
      <PriceField
        value={value}
        label="Lorem"
        hint="Ipsum"
        onChange={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with value, label and error", () => {
    const component = renderer.create(
      <PriceField
        value={value}
        label="Lorem"
        hint="Ipsum"
        error={true}
        onChange={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
