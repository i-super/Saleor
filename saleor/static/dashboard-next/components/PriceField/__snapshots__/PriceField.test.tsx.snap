// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceField /> renders with hint 1`] = `
<div
  className="Component-widgetContainer-4"
>
  <div
    className="Component-inputContainer-1"
  >
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13 Component-pullDown-2"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_min"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value=""
        />
        <span />
      </div>
      <p
        className="MuiFormHelperText-root-35"
      >
        Lorem ipsum
      </p>
    </div>
    <span
      className="Component-separator-3"
    >
      -
    </span>
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13 Component-pullDown-2"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_max"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value=""
        />
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`<PriceField /> renders with label 1`] = `
<div
  className="Component-widgetContainer-4"
>
  <div
    className="Component-inputContainer-1"
  >
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-29 MuiInputLabel-root-23 MuiInputLabel-formControl-24 MuiInputLabel-animated-27"
        data-shrink={false}
        htmlFor="price_min"
      >
        Lorem ipsum
      </label>
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_min"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value=""
        />
        <span />
      </div>
    </div>
    <span
      className="Component-separator-3"
    >
      -
    </span>
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13 Component-pullDown-2"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_max"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value=""
        />
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`<PriceField /> renders with label and hint 1`] = `
<div
  className="Component-widgetContainer-4"
>
  <div
    className="Component-inputContainer-1"
  >
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-29 MuiInputLabel-root-23 MuiInputLabel-formControl-24 MuiInputLabel-animated-27"
        data-shrink={false}
        htmlFor="price_min"
      >
        Lorem
      </label>
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_min"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value=""
        />
        <span />
      </div>
      <p
        className="MuiFormHelperText-root-35"
      >
        Ipsum
      </p>
    </div>
    <span
      className="Component-separator-3"
    >
      -
    </span>
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13 Component-pullDown-2"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_max"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value=""
        />
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`<PriceField /> renders with no value 1`] = `
<div
  className="Component-widgetContainer-4"
>
  <div
    className="Component-inputContainer-1"
  >
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13 Component-pullDown-2"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_min"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value=""
        />
        <span />
      </div>
    </div>
    <span
      className="Component-separator-3"
    >
      -
    </span>
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13 Component-pullDown-2"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_max"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value=""
        />
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`<PriceField /> renders with value 1`] = `
<div
  className="Component-widgetContainer-4"
>
  <div
    className="Component-inputContainer-1"
  >
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13 Component-pullDown-2"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_min"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value="10"
        />
        <span />
      </div>
    </div>
    <span
      className="Component-separator-3"
    >
      -
    </span>
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13 Component-pullDown-2"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_max"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value="30"
        />
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`<PriceField /> renders with value, label and error 1`] = `
<div
  className="Component-widgetContainer-4"
>
  <div
    className="Component-inputContainer-1"
  >
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-29 MuiFormLabel-error-32 MuiInputLabel-root-23 MuiInputLabel-formControl-24 MuiInputLabel-animated-27 MuiInputLabel-shrink-26"
        data-shrink={true}
        htmlFor="price_min"
      >
        Lorem
      </label>
      <div
        className="MuiInput-root-9 MuiInput-error-14 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13"
      >
        <input
          aria-invalid={true}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_min"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value="10"
        />
        <span />
      </div>
      <p
        className="MuiFormHelperText-root-35 MuiFormHelperText-error-36"
      >
        Ipsum
      </p>
    </div>
    <span
      className="Component-separator-3"
    >
      -
    </span>
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-9 MuiInput-error-14 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13 Component-pullDown-2"
      >
        <input
          aria-invalid={true}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_max"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value="30"
        />
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`<PriceField /> renders with value, label and hint 1`] = `
<div
  className="Component-widgetContainer-4"
>
  <div
    className="Component-inputContainer-1"
  >
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-29 MuiInputLabel-root-23 MuiInputLabel-formControl-24 MuiInputLabel-animated-27 MuiInputLabel-shrink-26"
        data-shrink={true}
        htmlFor="price_min"
      >
        Lorem
      </label>
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_min"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value="10"
        />
        <span />
      </div>
      <p
        className="MuiFormHelperText-root-35"
      >
        Ipsum
      </p>
    </div>
    <span
      className="Component-separator-3"
    >
      -
    </span>
    <div
      className="MuiFormControl-root-5"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-9 MuiInput-fullWidth-16 MuiInput-formControl-10 MuiInput-underline-13 Component-pullDown-2"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-17 MuiInput-inputType-21"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="price_max"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="number"
          value="30"
        />
        <span />
      </div>
    </div>
  </div>
</div>
`;
